### PR TITLE
 [Term Entry] Go Math Functions RoundToEven() #3223

### PR DESCRIPTION
**Pull request by alinasir85**
### Description

- Added the RoundToEven term for Go
- Closes Issue #3223 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Note these special cases:
+
+- Passing an already even integer will return the same even integer.
+- Passing `NaN` (Not-a-Number) will return `NaN`.
+- Passing positive or negative `Inf` will return value with the same sign as the argument.
+
+## Example
+
+In this Go code example, we have a simple program that rounds a float value to the nearest even integer, using the `RoundToEven()` function from the `math` package.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md:33]:
```suggestion
This Go code example demonstrates a basic implementations of the `RoundToEven()` function.
```

File: content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+"math"
+)
+
+func  main() {
+// Define a floating-point number
+x := 3.5
+
+// Round x to the nearest even integer
+rounded := math.RoundToEven(x)
+fmt.Printf("Rounded %v to the nearest even integer: %v\n", x, rounded)
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md:48]:
```suggestion
fmt.Printf("%v rounded to the nearest even integer yields: %v\n", x, rounded)
```

File: content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+}
+```
+
+The above code results in the following output:
+
+```shell
+Rounded 3.5 to the nearest even integer: 4
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md:55]: update to match edits above

File: content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+## Codebyte Example
+
+The example below is runnable and demonstrates the `RoundToEven()` function.
+
+```codebyte/golang
+
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/roundtoeven/roundtoeven.md:63]: remove the empty lines at the top and bottom of the code block

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@alinasir85 I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by caupolicandiaz** [comment]
@alinasir85 In the future please wait that a maintainer assigns the issue to you before submitting a PR

**Comment by alinasir85** [comment]
> @alinasir85 In the future please wait that a maintainer assigns the issue to you before submitting a PR

ok got it

**Comment by alinasir85** [comment]
> @alinasir85 I've added my initial comments. Please update the entry, and tag me when you've finished making changes. Thanks

@caupolicandiaz , suggested changes pushed

**Review by SSwiniarski — APPROVED** [review]
@alinasir85 @caupolicandiaz Looks good.

**Pull request merged**